scipy/stats/_rcont/rcont.pyx

```diff
@@ -1,18 +1,19 @@
 cimport numpy as np
 import numpy as np
-
-np.import_array()
-
 from numpy.random cimport bitgen_t
 from cpython.pycapsule cimport PyCapsule_GetPointer, PyCapsule_IsValid
 
+ctypedef np.int64_t tab_t
```

Comment:
I think this re-typedef is causing errors that you are seeing in the build logs. Moving this under the `cdef extern from "_rcont.h"` and using `from libc.stdint import int64_t` fixes the problem.

![image](https://user-images.githubusercontent.com/43181252/205611725-88141619-4422-459f-a1ce-5194e357db16.png)

Reply:
I am seeing the warnings again for a rebuild. Can't make sense of why it's happening for only `_rcont`.

Reply:
I am not a Cython expert, so I don't know whether this is an issue and/or what the correct way is. In my other projects, I wrap my C/C++ code with pybind11 where I know precisely what is happening. I still used Cython here to follow the general trend in Scipy.

Reply:
Can't we just silence the warning locally? We do that in Boost as a last resort when the warning is a false positive as it is here.

Reply:
> Can't we just silence the warning locally? We do that in Boost as a last resort when the warning is a false positive as it is here. If this were just C/C++, I would know how to do this, but here we have Cython in the mix and I don't know how to handle that.

I don't think it would be a good idea to silence the warning even though it's a false positive. Since it's just the Python header messing with the standard lib, we can move the include of `math.h` below other includes and that should do it. Look at this comment for details: https://github.com/scipy/scipy/pull/17357#discussion_r1040276694.